### PR TITLE
doc: Mesh DK prov handler review

### PR DESCRIPTION
**Pull request by greg-fer**
Review documentation and formatting.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: include/bluetooth/mesh/dk_prov.rst [hunk 1]
```diff
@@ -3,55 +3,51 @@
...
 It supports four types of Out Of Band (OOB) authentication methods and uses the Hardware Information driver to generate a deterministic UUID to uniquely represent the device.
+For more information about provisioning in Bluetooth Mesh, see the :ref:`zephyr:bluetooth_mesh_provisioning` page in Zephyr.
 
 Used primarily in sample applications, this handler acts as a reference implementation for the application-specific part of provisioning.
 
-The handling of the OOB authentication methods is closely tied to the hardware parameters of Nordic's Development Kits.
-For deployments with custom hardware, do not extend this module, but instead, create a similar custom module tailored specifically to your board capabilities.
+The handling of the OOB authentication methods is closely tied to the hardware parameters of the Development Kits from Nordic Semiconductor.
+For deployments with custom hardware, do not extend this module.
+Instead, create a similar custom module that is tailored specifically to your board capabilities.
```
Comment (greg-fer) [inline include/bluetooth/mesh/dk_prov.rst:14]: @trond-snekvik , is this something intuitive or we need to document this procedure of creating custom modules?

File: include/bluetooth/mesh/dk_prov.rst [hunk 2]
```diff
@@ -3,55 +3,51 @@
 Bluetooth Mesh provisioning handler for Nordic DKs
 ##################################################
 
-This application-side module is a basic implementation of provisioning process handling for Development Kits from Nordic Semiconductor.
+This application-side module is a basic implementation of the provisioning process handling for Development Kits from Nordic Semiconductor.
 It supports four types of Out Of Band (OOB) authentication methods and uses the Hardware Information driver to generate a deterministic UUID to uniquely represent the device.
+For more information about provisioning in Bluetooth Mesh, see the :ref:`zephyr:bluetooth_mesh_provisioning` page in Zephyr.
 
-Used primarily in sample applications, this handler acts as a reference implementation for the application-specific part of provisioning.
+Used primarily in :ref:`Bluetooth Mesh sample applications <ble_samples>`, this handler acts as a reference implementation for the application-specific part of provisioning.
```
Comment (greg-fer) [inline include/bluetooth/mesh/dk_prov.rst:10]: @trond-snekvik , how is the module configured to work with the samples? Is it automatically enabled? The sample Testing section focus on the app, and the provisioner handler is only briefly mentioned.

File: include/bluetooth/mesh/dk_prov.rst (hunk 1 above)
Comment (trond-snekvik) [reply include/bluetooth/mesh/dk_prov.rst:14]: Discussed offline, but here's my opinion: 
This is the way the api is supposed to be used, so I think that should be pretty clear. IMO we shouldn't add any guides or anything for this until we know we need it

File: include/bluetooth/mesh/dk_prov.rst (hunk 2 above)
Comment (trond-snekvik) [reply include/bluetooth/mesh/dk_prov.rst:10]: It is enabled through the `CONFIG_DK_PROV` option and through calling `bt_mesh_dk_prov_init()` in main.

File: include/bluetooth/mesh/dk_prov.rst (hunk 1 above)
Comment (greg-fer) [reply include/bluetooth/mesh/dk_prov.rst:14]: Agree.

**Pull request merged**